src/components/CreateBook.js

```diff
@@ -1,9 +1,41 @@
+import React, { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/books';
+import Input from './Input';
+
 function CreateBook() {
+  const [state, setState] = useState({ id: '', title: '', author: '' });
+  // function to store user inputs
+  const id = useSelector((state) => state.books).noOfBooks;
+  const onChange = (e) => {
+    setState({ ...state, id: id + 1, [e.target.name]: e.target.value });
+  };
```

Comment:
- [ Optional ] You did a great job, but I think there's a better way of generating a unique ID. This method may work at first but there are high risks of seeing many books with the same ID which may create bugs in the future. 
- **Suggestion**
generating ID based on date `Date.now()`
installing a random ID package `UUID`